themes/aura/assets/upsell.js

```diff
@@ -24,24 +13,45 @@ document.getElementById("upsell-form").addEventListener("submit", async function
   };
 
   formData.forEach((value, key) => {
-    if (key.startsWith("product_offers")) {
-      const productId = key;
-      upsellParams.product_offers[productId] = value;
+    if (!["upsell_id", "order_id"].includes(key)) {
+      upsellParams.product_offers[key] = value;
     }
   });
 
+  await submitAnswer(event, upsellParams);
```

Comment:
we can refactor this function so that it only handles the network part, changing the button state can be done right before it:
```suggestion
  setButtonIsLoading(event.target, true);
  await submitAnswer(event, upsellParams);
```